test/letters/containers/Main.unit.spec.jsx

```diff
@@ -1,55 +1,134 @@
     const tree = SkinDeep.shallowRender(<Main {...props}/>);
     expect(tree.subTree('#systemDownMessage')).to.be.ok;
   });
 
+  it('renders children for letter eligibility errors', () => {
+    const props = { ...defaultProps, lettersAvailability: 'letterEligibilityError', children };
+    const tree = SkinDeep.shallowRender(<Main {...props}/>);
+    const childText = tree.subTree('span').text();
+    expect(childText).to.equal(testText);
+  });
+
   it('should show letters unavailable message when service is unavailable', () => {
-    const props = _.merge({}, defaultProps, { lettersAvailability: 'unavailable' });
+    const props = { ...defaultProps, lettersAvailability: 'unavailable' };
     const tree = SkinDeep.shallowRender(<Main {...props}/>);
     expect(tree.subTree('#lettersUnavailable')).to.be.ok;
   });
+
+  it('renders system down message for all unspecified errors', () => {
+    const props = { ...defaultProps, lettersAvailability: 'bogusError' };
+    const tree = SkinDeep.shallowRender(<Main {...props}/>);
+    expect(tree.subTree('#systemDownMessage')).to.be.ok;
+  });
+
+  it('fetches letter data after mounting', () => {
+    const getLetterList = sinon.spy();
+    const props = { ...defaultProps, getLetterList };
+    const tree = SkinDeep.shallowRender(<Main {...props}/>);
+    const instance = tree.getMountedInstance();
+    instance.componentDidMount();
+    expect(getLetterList.callCount).to.equal(1);
+  });
+
+  it('fetches mailing address after mounting', () => {
+    const getMailingAddress = sinon.spy();
+    const props = { ...defaultProps, getMailingAddress };
+    const tree = SkinDeep.shallowRender(<Main {...props}/>);
+    const instance = tree.getMountedInstance();
+    instance.componentDidMount();
+    expect(getMailingAddress.callCount).to.equal(1);
+  });
+
+  it('fetches benefit summary options after mounting', () => {
+    const getBenefitSummaryOptions = sinon.spy();
+    const props = { ...defaultProps, getBenefitSummaryOptions };
+    const tree = SkinDeep.shallowRender(<Main {...props}/>);
+    const instance = tree.getMountedInstance();
+    instance.componentDidMount();
+    expect(getBenefitSummaryOptions.callCount).to.equal(1);
+  });
+
+  it('fetches country list after mounting', () => {
+    const getAddressCountries = sinon.spy();
+    const props = { ...defaultProps, getAddressCountries };
+    const tree = SkinDeep.shallowRender(<Main {...props}/>);
+    const instance = tree.getMountedInstance();
+    instance.componentDidMount();
+    expect(getAddressCountries.callCount).to.equal(1);
+  });
+
+  it('fetches state list after mounting', () => {
+    const getAddressStates = sinon.spy();
+    const props = { ...defaultProps, getAddressStates };
+    const tree = SkinDeep.shallowRender(<Main {...props}/>);
+    const instance = tree.getMountedInstance();
+    instance.componentDidMount();
+    expect(getAddressStates.callCount).to.equal(1);
```

Comment:
I wonder if it'd make sense to bundle these tests into a single one that tests that `it('should fetch all the things after mounting')`. That way, we would only shallow render once instead of five times. If the test fails, we'd see which line failed, so we'd know what needs to be fixed, but...it may be a micro-optimization. I can see it going either way.

Reply:
Yeah, I was on the fence about this and actually initially wrote them this way. At some point I thought it might be simpler to just leave them separated but with you weighing in I'm leaning toward keeping things DRY.